webapp/src/shared/components/DispatchPage/DispatchPage.tsx

```diff
@@ -20,7 +30,33 @@ export default function DispatchPage<T>({
   dataMapper,
   button,
 }: DispatchPageProps<T>) {
-  const [data, setData] = useState<T[]>([]);
+  const [search, setSearch] = useState('');
+  const [offset, setOffset] = useState(0);
+  const getData = useCallback(() => {
+    return fetchFn({ search, offset });
+  }, [fetchFn, search, offset]);
+  const { data, isLoading, hasMore } = UseSearch(search, getData);
+  const observer = useRef<IntersectionObserver | null>(null);
+
+  const lastRowElementRef = useCallback(
+    (row: HTMLLIElement) => {
+      if (isLoading) {
+        return;
+      }
+      if (observer.current) {
+        observer.current.disconnect();
+      }
+      observer.current = new IntersectionObserver((entries) => {
```

Comment:
It's a bit hard to follow. I'm assuming you are creating a last row item with a special behaviour, but I'm not sure yet. Do you have an easy way to test it or should I manually add dozens of users to the database until some offset and limit are meaningful?

Reply:
How it works:

We assign a callback ref to the last row item. React will call that callback whenever the ref gets attached to a different row. When the component mounts, it calls the callback with the element, and when it unmounts, it calls with null as the parameter.

This article helped me to understand how callback refs work: https://tkdodo.eu/blog/avoiding-use-effect-with-callback-refs.

Inside the callback ref, we create an `IntersectionObserver` and set it to observe the last row item.
The observer also has a callback and calls it when an observed element intersects with the viewport (when it appears or disappears from the screen).

```tsx
(entries) => {
        if (entries[0].isIntersecting && hasMore) {
          setOffset(data.length);
        }
      }
```

When the `offset` changes, the `getData` method updates, which triggers the `useEffect` from the `useData` hook.

I hope this helps, and sorry, it is a bit confusing.

Reply:
> Do you have an easy way to test it or should I manually add dozens of users to the database until some offset and limit are meaningful?

We could create a script to seed the database with random users. I will look into it.

Reply:
To test this until we implement a script to seed the database, please go to https://www.mockaroo.com/ea0e3c90. Download the `SeedUsers.sql` file, copy the file to db/setup/, then execute:
```sh
docker compose down -v && docker compose up -d --build -V
```

This will insert 1000 random users to the Users table

Reply:
as an extract of the errors when running the commands you mention I have this:

```
ft_transcendence-db-1                 | psql:/var/lib/postgresql/setup/SeedUsers.sql:987: ERROR:  relation "seedusers" does not exist
ft_transcendence-db-1                 | LINE 1: insert into SeedUsers ("username", "fullName", "email") valu...
```

Reply:
(I could fix it by changing the table name to users in mockaroo, before downloading the file, but maybe this is not the way you intended)

Reply:
You are right. The table name should be `users`. Now it's fixed.